pkg/prometheus/client.go

```diff
@@ -53,6 +57,15 @@ type ClientIn struct {
 }
 
 func providePrometheusClient(in ClientIn) (prometheusv1.API, promapi.Client, error) {
+	var fluxNinjaConfig extconfig.FluxNinjaExtensionConfig
+	if err := in.Unmarshaller.UnmarshalKey(fluxninjaCOnfigKey, &fluxNinjaConfig); err != nil {
+		log.Error().Err(err).Msg("unable to deserialize")
```

Comment:
Nit: Either change to warn and be more descriptive what fails, or just ignore the error altogether